cmd/nerdctl/container_create_windows_test.go

```diff
@@ -0,0 +1,70 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"testing"
+	"time"
+
+	"github.com/containerd/nerdctl/pkg/testutil"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+func TestCreateProcessContainer(t *testing.T) {
```

Comment:
Why do we need a separate set of tests? Can we not reuse the same tests and mark windows incompatible tests and windows only tests in testutil like docker

Reply:
The [`--isolation` flag](https://github.com/containerd/nerdctl/pull/2145/files/a176d8cca3cc9cfae0f10f3abc5a872dc75c6ec6#diff-aa8815c52e48ac75edf93e1ba0ef7a71e3d16bec6b485505a9d614b0ae2863c2R60) mandates the need for at least some dedicated tests on Windows.

I agree that `TestCreateProcessContainer` should explicitly set `--isolation process` though, so maybe the actual testing logic in this and `TestCreateHyperVContainer` could be abstracted into a testing helper which takes as an argument the isolation type it should use? @dardelean

Reply:
Can you see `testutil.DockerIncompatible(t)` and implement something as `testutil.DockerHyperVCompatible(t)` and run those specific tests when this flag is set. You would probably need testutl_linux.go that basically always skips the test running on a linux machine.